### PR TITLE
Use terraform-state-read-role-tf-module

### PR DESCRIPTION
**Pull request by dav3r**
#  #

## 🗣 Description ##

This PR uses the new [`cisagov/terraform-state-read-role-tf-module`](https://github.com/cisagov/terraform-state-read-role-tf-module) to create the role and policies for read-only access to this project's Terraform state.

## 💭 Motivation and context ##

Using a Terraform module simplifies the code in this repository and simpler code is better code.

## 🧪 Testing ##

I successfully applied these changes in Staging and verified that they produced the expected output.  

Note: Before the apply could be done, I had to manually detach the existing `AssumeReadSharedServicesNetworkingTerraformState` policy from every group it was attached to (the PCA and Domain Manager provisioner groups), since a policy that is attached to an entity cannot be deleted (which had to be done as part of this apply).  After the apply finished, I attached the newly-re-created `AssumeReadSharedServicesNetworkingTerraformState` policy to the groups that I had detached it from earlier.

## ✅ Checklist ##

* [x] This PR has an informative and human-readable title.
* [x] Changes are limited to a single goal - _eschew scope creep!_
* [x] All relevant type-of-change labels have been added.
* [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
* [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
* [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
* [x] All new and existing tests pass.

## ✅ Post-Merge Checklist ##
* [x] Apply this change in Production (will have to use `terraform import` for the IAM resources that are shared between environments)

### CODE REVIEW

File: outputs.tf [hunk 1]
```diff
@@ -48,9 +43,9 @@ output "public_subnets" {
   description = "The public subnets in the VPC."
 }
 
-output "read_terraform_state_role" {
-  value       = aws_iam_role.read_terraform_state
-  description = "The role that allows read-only access to the cool-sharedservices-networking state in the Terraform state bucket."
+output "read_terraform_state" {
+  value       = module.read_terraform_state
+  description = "The IAM policies and role that allow read-only access to the cool-sharedservices-networking state in the Terraform state bucket."
```
Comment (jsf9k) [inline outputs.tf:48]: It never occurred to me to output an entire module.  C'est intéressant.

**Review by jsf9k — APPROVED** [review]
I love it!

File: outputs.tf (hunk 1 above)
Comment (dav3r) [reply outputs.tf:48]: I didn't see any downsides to doing it this way, so I figured I'd give it a shot.

**Review by mcdonnnj — APPROVED** [review]
Love the new look! ✨

**Pull request merged**